src/main/java/seedu/academydirectory/logic/commands/RetrieveCommand.java

```diff
@@ -31,33 +32,53 @@
             + "\n"
             + "Format: `retrieve INFORMATION [of STUDENT_NAME]`";
 
-    // TODO: Support for individual names
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Retrieve personal details of students "
             + "and displays them as a list\n"
             + "Parameters: " + PREFIX_EMAIL + " | "
             + PREFIX_TELEGRAM + " | " + PREFIX_PHONE + "\n"
             + "Example: " + COMMAND_WORD + " " + PREFIX_PHONE;
 
-    private final Function<? super Student, ? extends Information> filter;
+    private final List<InformationWantedFunction> filterList;
+
+    public RetrieveCommand(List<InformationWantedFunction> filterList) {
+        this.filterList = filterList;
+    }
 
-    public RetrieveCommand(Function<? super Student, ? extends Information> filter) {
-        this.filter = filter;
+    public RetrieveCommand(InformationWantedFunction filter) {
+        this(List.of(filter));
+    }
+
+    private String executeFilter(Model model, InformationWantedFunction filter) {
+        ObservableList<Information> view = model.getFilteredStudentListView(filter)
+                .stream()
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toCollection(FXCollections::observableArrayList));
+
+        return view.size() == 0
+                ? String.format(Messages.MESSAGE_STUDENTS_LISTED_OVERVIEW, 0)
+                : view.stream().map(Object::toString).collect(Collectors.joining("\n"));
     }
 
```

Comment:
Do you want to add JavaDoc to these methods?
Also I felt that the code is a bit too complicated - granted that you are using functional programming, but may consider making it a bit simpler later on? Just use direct if-else branch.

Reply:
Alright. Thanks!